translator/src/translator.rs

```diff
@@ -0,0 +1,450 @@
+            Expression::SetComp(s) => self.visit_set_comp(s),
+            Expression::DictComp(d) => self.visit_dict_comp(d),
+            Expression::Attribute(a) => self.visit_attribute(a),
+            Expression::Subscript(s) => self.visit_subscript(s),
+            Expression::Slice(s) => self.visit_slice(s),
+            Expression::Call(c) => self.visit_call(c),
+            Expression::Await(a) => self.visit_await(a),
+            Expression::Compare(c) => self.visit_compare(c),
+            Expression::Lambda(l) => self.visit_lambda(l),
+            Expression::IfExp(i) => self.visit_if_exp(i),
+            Expression::JoinedStr(j) => self.visit_joined_str(j),
+            Expression::FormattedValue(f) => self.visit_formatted_value(f),
+        }
+    }
+
+    fn visit_constant(&mut self, constant: &Constant) {
+        match constant.value {
+            ConstantValue::None => write!(self.output, "None"),
+            ConstantValue::Ellipsis => write!(self.output, "..."),
+            ConstantValue::Bool(_) => write!(self.output, "bool"),
+            ConstantValue::Str(_) => write!(self.output, "\"{}\"", constant.get_value(&self.file.source).to_string()),
+            ConstantValue::Bytes => write!(self.output, "bytes"),
+            ConstantValue::Tuple => write!(self.output, "tuple"),
+            ConstantValue::Int => write!(self.output, "{}", constant.get_value(&self.file.source).to_string()),
+            ConstantValue::Float => write!(self.output, "{}", constant.get_value(&self.file.source).to_string()),
+            ConstantValue::Complex => write!(self.output, "complex"),
+            /*
+            Constant::Tuple(elements) => {
+                let tuple_elements: Vec<String> = elements
+                    .iter()
+                    .map(|elem| self.translate_constant(elem))
+                    .collect::<Result<Vec<String>, _>>()?;
+                Ok(format!("({})", tuple_elements.join(", ")))
+            },
+            */
+        };
+    }
+
+    fn visit_import(&mut self, import: &Import) {
+        for name in import.names.iter() {
+            if name.name == "torch" {
+                writeln!(self.output, "#include <torch/torch.h>");
+            }
+        }
+    }
+
+    fn visit_assign(&mut self, a: &Assign) {
+        let symbol_table = self.checker.get_symbol_table(None);
+        for target in &a.targets {
+            // let type = self.checker.types.
+            match target {
+                Expression::Name(n) => {
+                    let node = symbol_table.lookup_in_scope(&n.id, self.current_scope);
+                    match node {
+                        Some(node) => {
+                            let path = node.declarations[0].declaration_path();
+                            if path.node == n.node {
+                                let typ = self.checker.get_type(&n.node);
+                                write!(self.output, "{} ", self.python_type_to_cpp(&typ));
```

Comment:
maybe this can be a visit_type()? looks like we need it in many other places too

Reply:
This is an interesting point -- the visit_{entity} functions are just for all the nodes in the Python parse tree (like expression, statement, comment) and types in general don't correspond to entities of the parse tree, unless they are annotations, but not in general -- that's why there is no visit_type function :)

Reply:
oh, then maybe not as part of the visit, maybe something like `annotate_type`? it's re-used everywhere!